ciao-controller/openstack_compute.go

```diff
@@ -74,6 +77,234 @@ func instanceToServer(ctl *controller, instance *types.Instance) (compute.Server
 	return server, nil
 }
 
+func validateBlockDeviceMappingSourceType(srcType string) (err error) {
+	validSourceTypes := []string{
```

Comment:
a minor style preference, for return value of error here where you never actually use the variable named "err" in the function, I would prefer to leave return value unnamed. it is a trivail thing of course, but it does prevent an unneeded allocation and initialization of "err" since you aren't using it, and naming error "err" doesn't actually provide any additional readability information.